ui/site/src/socket.js

```diff
@@ -65,14 +65,15 @@ lichess.StrongSocket = function(url, version, settings) {
   };
 
   var send = function(t, d, o, noRetry) {
-    d = d == null ? {} : d,
-      o = o || {};
-    if (o.withLag) d.l = Math.round(averageLag);
+    o = o || {};
+    var msg = { t: t };
+    if (d !== undefined) {
+      if (o.withLag) d.l = Math.round(averageLag);
+      if ('millis' in o) d.s = Math.floor(o.millis * 0.1).toString(36);
```

Comment:
Direct property access is normally optimized by the engine. I think the `in` operator can't be optimized.
The other problem is that `in` doesn't tell you the field is defined, it just tells you that the key 'millis' exists. So with that check you could still send `NaN` (or `0` if millis is null) to the server.